data_validation/config_manager.py

```diff
@@ -763,14 +784,12 @@ def append_pre_agg_calc_field(
                 self.source_client.name == "bigquery"
                 or self.target_client.name == "bigquery"
             ):
-                calc_func = "cast"
-                cast_type = "timestamp"
                 pre_calculated_config = self.build_and_append_pre_agg_calc_config(
                     source_column,
                     target_column,
-                    calc_func,
+                    "cast",
                     column_position,
-                    cast_type,
+                    "timestamp",
                     depth,
                 )
                 source_column = target_column = pre_calculated_config[
```

Comment:
It works from right to left. So target_column is set to the right most expression and then source_column is set to the value of target_column. I don't often use this technique because it is easy to not see the middle variable, but it is valid and sometimes makes sense.

Reply:
Ahh got it, that's very cool! Thanks for explaining 🙏 I'll be resolving this comment!